src/Microsoft.ML.Data/DataLoadSave/Binary/CodecFactory.cs

```diff
@@ -68,7 +68,8 @@ public CodecFactory(IHostEnvironment env, MemoryStreamPool memPool = null)
             RegisterOtherCodec("DvTimeSpan", new UnsafeTypeCodec<TimeSpan>(this).GetCodec);
 
             RegisterOtherCodec("VBuffer", GetVBufferCodec);
-            RegisterOtherCodec("Key", GetKeyCodec);
+            RegisterOtherCodec("KeyNew", GetKeyCodec);
```

Comment:
>KeyNew [](start = 32, length = 6)

"New" is relative. Everything here is old but once was new... Maybe Key2? Otherwise if a new version comes along, we'd have "Key", "KeyNew", and then, maybe, "KeySuperNew".

Reply:
Well I don't think this protocol was very well designed to begin with as it does not allow for a version number. I was trying to reflect the practice in the rest of the file where there seems to be a "Old" version of some codecs which says "old" in their name. 

I agree however that your suggestion of numbering would be the least harm here.